state-sync/state-sync-driver/src/driver.rs

```diff
@@ -234,7 +234,7 @@ impl<
     async fn handle_consensus_notification(&mut self, notification: ConsensusNotification) {
         // Verify the notification: full nodes shouldn't receive notifications
         // and consensus should only send notifications after bootstrapping!
-        let result = if self.driver_configuration.role == RoleType::FullNode {
+        let result = if !self.is_validator() {
```

Comment:
Maybe add a new function, e.g., `is_consensus_running()`, and then check for consensus or consensus observer? 😄